src/sentry/runner/commands/cluster_transaction_names.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from datetime import datetime, timedelta
+from typing import List, Literal, Mapping, Tuple, TypedDict
+
+import click
+from snuba_sdk import Column, Condition, Entity, Op, Query, Request
+
+from sentry.ingest.transaction_clusterer.base import ReplacementRule
+from sentry.runner.decorators import configuration
+
+logger = logging.getLogger(__name__)
+
+
+RULE_RETENTION = timedelta(days=90)
+OPTION_NAME = "sentry:transaction_name_normalize_rules"
+
+
+class RuleSpec(TypedDict):
+    type: Literal["glob"]
+    value: ReplacementRule
+    expires: int  # unix timestamp
```

Comment:
This format is open for discussion, I just came up with something, inspired by dynamic sampling rules.